modules/core/src/test/java/org/apache/ignite/internal/GridCheckpointTest.java

```diff
@@ -0,0 +1,158 @@
+ */
+public class GridCheckpointTest extends GridCommonAbstractTest {
+    /** */
+    public GridCheckpointTest() {
+        super(/*start Grid*/false);
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void beforeTest() throws Exception {
+        stopAllGrids();
+
+        cleanPersistenceDir();
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void afterTest() throws Exception {
+        stopAllGrids();
+
+        cleanPersistenceDir();
+    }
+
+    /** */
+    @Override protected IgniteConfiguration getConfiguration(String gridName) throws Exception {
+        IgniteConfiguration cfg = super.getConfiguration(gridName);
+
+        cfg.setConsistentId(gridName);
+
+        CacheConfiguration<Object, Object> ccfg1 = new CacheConfiguration<>(DEFAULT_CACHE_NAME)
+            .setAtomicityMode(CacheAtomicityMode.ATOMIC)
+            .setCacheMode(CacheMode.PARTITIONED)
+            .setBackups(1)
+            .setAffinity(new RendezvousAffinityFunction(false, 16));
+
+        cfg.setCacheConfiguration(ccfg1);
+
+        DataStorageConfiguration dbCfg = new DataStorageConfiguration()
+            .setWalMode(WALMode.LOG_ONLY)
+            .setWalSegmentSize(1024 * 1024)
+            .setCheckpointFrequency(Integer.MAX_VALUE)
+            .setWalCompactionEnabled(true)
+            .setDefaultDataRegionConfiguration(
+                new DataRegionConfiguration()
+                    .setPersistenceEnabled(true)
+                    .setMaxSize(100 * 1024 * 1024)
+            );
+
+        cfg.setDataStorageConfiguration(dbCfg);
+
+        return cfg;
+    }
+
+    /** */
+    protected CacheConfiguration cacheConfiguration() {
+        CacheConfiguration cc = defaultCacheConfiguration();
+        cc.setName(DEFAULT_CACHE_NAME);
+
+        return cc;
+    }
+
+    @Test
+    public void testForceCheckpointing() throws Exception {
+        IgniteEx ignite1 = startGrid(1);
+
+        final GridCacheDatabaseSharedManager gridDb = (GridCacheDatabaseSharedManager)ignite1.context().cache().context().database();
+        final CheckpointHistory checkpointHist = gridDb.checkpointHistory();
+        ignite1.configuration().getCheckpointSpi();
+        assert null != checkpointHist;
+
+        final int numOfCheckpointsBefore = checkpointHist.checkpoints().size();
+        final CheckpointEntry lastCheckpointBefore = checkpointHist.lastCheckpoint();
+
+        ignite1.cluster().state(ClusterState.ACTIVE);
+
+        IgniteCache<Object, Object> jcache = jcache(1);
+        jcache.put(1, 1);
+
+        ignite1.cluster().checkpoint();
+
+        final int numOfCheckpointsAfter = checkpointHist.checkpoints().size();
+        final CheckpointEntry lastCheckpointAfter = checkpointHist.lastCheckpoint();
+
+        assertTrue(numOfCheckpointsAfter > numOfCheckpointsBefore);
+        assertNotSame(lastCheckpointBefore, lastCheckpointAfter);
+    }
+
+    @Test
+    public void testForceCheckpointingJmx() throws Exception {
```

Comment:
Do we need an additional test for a case when cluster does not contains any server node?